shpkpr/exceptions.py

```diff
@@ -0,0 +1,42 @@
+# stdlib imports
+import sys
+
+# third-party imports
+import six
+from click.exceptions import ClickException
+
+
+class ShpkprException(ClickException):
+    """Common base class for all exceptions raised explicitly by shpkpr.
+
+    Exceptions which are subclasses of this type will be handled nicely by
+    shpkpr and a human-readable error message will be displayed to the user
+    before exiting with an appropriate non-zero exit code. Any exceptions
+    raised which are not a subclass of this type will exit(1) and print a
+    traceback to the user's console.
+    """
+    pass
```

Comment:
What about implementing the __init__ method with a NotImplemented exception to prevent people from initialising and hiding away errors? I don't think you're going to making a super call